### PR TITLE
ci: use github token permissions instead of personal access tokens

### PR DESCRIPTION
**Pull request by nunofgs**
### Description

Replaces PATs currently being used for various CI actions with `GITHUB_TOKEN` and respective [permissions](https://github.blog/changelog/2021-04-20-github-actions-control-permissions-for-github_token/).

These changes match the work being done on https://github.com/customerio/customerio-ios/pull/261.

### Note to reviewers
<details>
  <summary>The `deploy-npm` and `deploy-git-tag` workflows were merged into a single workflow.</summary>
  <br/>
  <ul>
   <li>This was required because events generated by the use of `GITHUB_TOKEN` <a href="https://docs.github.com/en/actions/using-workflows/triggering-a-workflow#triggering-a-workflow-from-a-workflow">will not trigger</a> other workflows.</li>
  <li>
  This will be the new flow:
  Screenshot 2023-02-10 at 12 51 30

  </li>
</details>

<details>
  <summary>The `action-semantic-release-justwork` step was removed.</summary>
  <br/>
  <ul>
   <li>This is no longer necessary since we're not updating long-running tags anymore. (such as `v1`, `v2`, etc.)</li>
  </ul>
</details>

<details>
  <summary>Pull requests are no longer created as `develop` branch is being phased out.</summary>
  <br/>
  <ul>
   <li>Previously this was only done if the automatic merge failed.</li>
  </ul>
</details>

<details>
  <summary>actions/checkout was bumped from v2 to v3.</summary>
  <br/>
  <ul>
   <li>Just cleanup!</li>
  </ul>
</details>

### CODE REVIEW

File: .github/workflows/deploy-sdk.yml [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+
+# Only run after a pull request has been merged. This is because
+# bot account runs write operations on the github repo to push a tag.
+on:
+  push:
+    branches: [main] # all branches where deployments currently occur. Make sure this list matches list of branches in  `.releaserc` file.
+
+permissions:
+  contents: write # access to push the git tag
+  issues: write # Bot creates an issue if there is an issue during deployment process
```
Comment (levibostian) [inline .github/workflows/deploy-sdk.yml:11]:
```suggestion
  issues: write # Bot creates an issue if there is an issue during deployment process
  pull-requests: write # allow bot to make comments on PRs after they get deployed
```

File: .github/workflows/deploy-sdk.yml [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+              ]
+            }
+        env:
+          SLACK_WEBHOOK_URL: ${{ secrets.SLACK_WEBHOOK_URL }}
+          SLACK_WEBHOOK_TYPE: INCOMING_WEBHOOK
+
+  deploy-npm:
+    name: Deploy to npm
+    needs: [deploy-git-tag]
+    if: needs.deploy-git-tag.outputs.new_release_published == 'true' # only run if a git tag was made.
```
Comment (levibostian) [inline .github/workflows/deploy-sdk.yml:109]: the `deploy-git-tag` job needs to have `outputs` setup in order for this to work.

File: .github/workflows/deploy-sdk.yml (hunk 1 above)
Comment (nunofgs) [reply .github/workflows/deploy-sdk.yml:11]: Pull requests are no longer being created in this workflow and, AFAIK, neither are comments. Do you still see a need for it?

**Review by levibostian — APPROVED** [review]

**Comment by levibostian** [comment]
Sorry for the git commit and revert. GitHub wasn't showing me latest changes for some reason so I made a change I was predicting you would make. 

We're good to go now 👍

**Review by levibostian — APPROVED** [review]

**Pull request merged**